src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -474,9 +471,9 @@ internal static IEnumerable<IEdmFunction> RemoveActions(this IEnumerable<IEdmOpe
         /// <param name="source">The source.</param>
         /// <param name="actionImportItems">The action import items.</param>
         /// <returns>Only the function imports from the operation Import sequence.</returns>
-        internal static IEnumerable<IEdmFunctionImport> RemoveActionImports(this IEnumerable<IEdmOperationImport> source, out IList<IEdmActionImport> actionImportItems)
+        internal static IEnumerable<IEdmOperationImport> RemoveActionImports(this IEnumerable<IEdmOperationImport> source, out IList<IEdmActionImport> actionImportItems)

```

Comment:
>IEdmOperationImport [](start = 36, length = 19)

Why make the return type IEdmOperationImport, rather than keep it as IEdmFunctionImport, since the returned values are always function imports?

Reply:
Because the return will be used to call "FilterOperationsByParameterNames", which accepts the "IEdmOperation". We don't want to cast the operation to function, then cast the function to operation again and again. That's useless and waste of time. 

Check the line 45 in FunctionOverloadResolver.cs.

---
In reply to: [289996208](https://github.com/OData/odata.net/pull/1473#discussion_r289996208) [](ancestors = 289996208)

Reply:
We don't have to cast because we are passing a more derived type. Passing a derived type to a method that is typed to receive the base type should be free, as the type checking is done at compile time, not at execution.

---
In reply to: [290003738](https://github.com/OData/odata.net/pull/1473#discussion_r290003738) [](ancestors = 290003738,289996208)

Reply:
As we discussed, since:
1) .NET 3.5 requires an explicit cast to cast from IEnumerable<type> to IEnumerable<derivedType>
2) The methods that currently take the returned enumerables treat them as operations, and never have to cast to function/action
3) The current code in EdmLibraryExtensions *does* cast them to the correct derived type (even though downstream we only care about the base type), and
4) These methods are all internal, so we could change them later to return the derived type *if* we had a need to treat them as the derived type
Then I'm okay, for now, changing all of the signatures to deal with operations, rather than the appropriate derived type. 

---
In reply to: [290046571](https://github.com/OData/odata.net/pull/1473#discussion_r290046571) [](ancestors = 290046571,290003738,289996208)